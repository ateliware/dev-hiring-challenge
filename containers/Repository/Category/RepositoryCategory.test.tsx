import {
  fireEvent,
  render,
  screen,
  waitForElementToBeRemoved,
} from "@testing-library/react";

import { RepositoryCategory } from ".";

describe("RepositoryCategory", () => {
  it("renderiza o grupo de repositorios", async () => {
    const wrapper = render(
      <RepositoryCategory name="Javascript" language="JavaScript" query="" />
    );
    //verificar se está mostrando o texto de carregando enquando se obtém os repositories
    const loadingMsg = screen.getByText("carregando...");
    expect(loadingMsg).toBeInTheDocument();
    const categoryTitle = screen.getByRole("heading", { name: "Javascript" });
    expect(categoryTitle).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingMsg);
    //verificar se uma vez que a mensagem de carregando sai a lista de repositorios é renderizada
    expect(screen.getByRole("heading", { name: "vue" })).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: "prettier" })
    ).toBeInTheDocument();
    const refreshResults = screen.getByText(
      "Atualizar banco de dados desde a API de Github",
      { selector: "button" }
    );
    expect(refreshResults).toBeInTheDocument();
    //simular click em atualizar resultados desde a API de Github
    fireEvent.click(refreshResults);
    //verificar se existe a mensagem de "atualizando"
    const updatingMsg = screen.getByText("atualizando...", { selector: "div" });
    expect(updatingMsg).toBeInTheDocument();
    await waitForElementToBeRemoved(updatingMsg);
    //verificar se uma vez que a mensagem de atualizando sai a lista de repositorios é renderizada
    expect(screen.getByRole("heading", { name: "vue" })).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: "prettier" })
    ).toBeInTheDocument();
    expect(wrapper.container).toMatchSnapshot();
  });
  it("mostra mensagem de erro quando tem problema com a API", async () => {
    const wrapper = render(
      <RepositoryCategory name="Typescript" language="TypeScript" query="" />
    );
    const loadingMsg = screen.getByText("carregando...");
    expect(loadingMsg).toBeInTheDocument();
    const categoryTitle = screen.getByRole("heading", { name: "Typescript" });
    expect(categoryTitle).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingMsg);
    const alertMsg = screen.getByRole("alert");
    expect(alertMsg).toHaveTextContent(
      "Ops! aconteceu um erro consultando os repositórios."
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
