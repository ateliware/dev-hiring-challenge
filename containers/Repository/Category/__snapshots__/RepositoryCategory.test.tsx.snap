// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCategory mostra mensagem de erro quando tem problema com a API 1`] = `
<div>
  <article
    class="main"
  >
    <header
      class="header"
    >
      <h3
        class="title"
      >
        Typescript
      </h3>
      <p
        class="lead"
      >
        Mostrando 0 repósitorios, ordenados por quantidade de estrelas do servidor MySQL local atualizado em 
      </p>
      <button
        class="refreshDataButton"
      >
        Atualizar banco de dados desde a API de Github
      </button>
    </header>
    <main
      class="alertContainer"
    >
      <div
        class="main error"
        role="alert"
      >
        <main>
          Ops! aconteceu um erro consultando os repositórios.
           
        </main>
        <footer
          class="footer"
        >
          <hr
            class="hr"
          />
          <ul
            class="actions"
          >
            <li>
              <button>
                tentar novamente
              </button>
            </li>
          </ul>
        </footer>
      </div>
    </main>
  </article>
</div>
`;

exports[`RepositoryCategory renderiza o grupo de repositorios 1`] = `
<div>
  <article
    class="main"
  >
    <header
      class="header"
    >
      <h3
        class="title"
      >
        Javascript
      </h3>
      <p
        class="lead"
      >
        Mostrando 2 repósitorios, ordenados por quantidade de estrelas do servidor MySQL local atualizado em 14 de fevereiro de 2022 20:09:47
      </p>
      <button
        class="refreshDataButton"
      >
        Atualizar banco de dados desde a API de Github
      </button>
    </header>
    <ul
      class="main"
    >
      <li
        class="main"
      >
        <header
          class="header"
        >
          <a
            href="https://github.com/vuejs/vue"
            rel="noreferrer"
            target="_blank"
          >
            <h4
              class="title"
            >
              vue
            </h4>
          </a>
        </header>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Estrelas
          </span>
          <span
            class="sectionValue"
          >
            193036
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Proprietario
          </span>
          <span
            class="sectionValue"
          >
            <div
              class="ownerInfo"
            >
              <img
                alt="vuejs"
                src="https://avatars.githubusercontent.com/u/6128107?v=4"
              />
              <span>
                vuejs
              </span>
            </div>
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Descrição
          </span>
          <span
            class="sectionValue"
          >
            🖖 Vue.js is a progressive, incrementally-adoptable JavaScript framework for building UI on the web.
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Tópicos
          </span>
          <span
            class="sectionValue"
          >
            framework, frontend, javascript, vue
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Linguagem
          </span>
          <span
            class="sectionValue"
          >
            JavaScript
          </span>
        </section>
      </li>
      <li
        class="main"
      >
        <header
          class="header"
        >
          <a
            href="https://github.com/prettier/prettier"
            rel="noreferrer"
            target="_blank"
          >
            <h4
              class="title"
            >
              prettier
            </h4>
          </a>
        </header>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Estrelas
          </span>
          <span
            class="sectionValue"
          >
            41888
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Proprietario
          </span>
          <span
            class="sectionValue"
          >
            <div
              class="ownerInfo"
            >
              <img
                alt="prettier"
                src="https://avatars.githubusercontent.com/u/25822731?v=4"
              />
              <span>
                prettier
              </span>
            </div>
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Descrição
          </span>
          <span
            class="sectionValue"
          >
            Prettier is an opinionated code formatter.
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Tópicos
          </span>
          <span
            class="sectionValue"
          >
            angular, ast, css, flow, formatter, graphql, html, javascript, json, jsx, less, markdown, prettier, printer, scss, typescript, vue, yaml
          </span>
        </section>
        <section
          class="section"
        >
          <span
            class="sectionLabel"
          >
            Linguagem
          </span>
          <span
            class="sectionValue"
          >
            JavaScript
          </span>
        </section>
      </li>
    </ul>
  </article>
</div>
`;
